internal/wasm/binary/value.go

```diff
@@ -49,5 +55,8 @@ func decodeUTF8(r *bytes.Reader, contextFormat string, contextArgs ...interface{
 		return "", 0, fmt.Errorf("%s is not valid UTF-8", fmt.Sprintf(contextFormat, contextArgs...))
 	}
 
-	return string(buf), size + uint32(sizeOfSize), nil
+	// TODO: use unsafe.String after flooring 1.20.
+	ret := *(*string)(unsafe.Pointer(&buf))
+	runtime.KeepAlive(buf)
```

Comment:
I don't think this is necessary, the compiler can track the slice pointer being transferred to the string.

Reply:
cool thanks!